File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.go
func minPairSum(nums []int) int {
    
    sort.Ints(nums)
    
    n:= len(nums)
    result := 0
    
    for i:= 0; i < n/2 ; i++  {
        result = max(result, nums[i] + nums[n - i - 1])
    }
    return result
}

File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.py
class Solution(object):
    def minPairSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums.sort()
        n = len(nums)
        
        result = 0
        
        for i in range(n//2):
            result = max(result, nums[i] + nums[n - i - 1])
        
        return result